### DISCUSSION TITLE
custom domain for only one pages project

### DISCUSSION THREAD

**Discussion by dangrunebaum** [community/community#63587] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

Is it possible to have a custom domain for only one github pages project? For example, I would like to use the custom domain "mycustomdomain.org" for "myname.github.io/myproject" but NOT affect the URLs of existing pages i.e. "myname.github.io/myotherprojects" Thanks!

**Comment by daneedev** [comment]
Hey,
of course. You can assign different domain or subdomain for each project, as example you can solve it with subdomains: 

- First project: **project1.mydomain.com**
- Second project **project2.mydomain.com**

## How to do that
1. Go to your repository and click on github pages settings
2. In Custom domain field write your domain
3. In your DNS provider add new record like this:

|  Name  |  Record Type  |  Content |
| ---- | --- | --- |
|  project1.yourdomain.com |  CNAME  |  dangrunebaum.github.io  |
|  project2.yourdomain.com |  CNAME  |  dangrunebaum.github.io  |

**Reply by TheDogHusky** [reply]
Possible to just use a custom domain for one pages, and leave the other with github's domain?
So one would be mysite.example.com but the others would still be me.github.io/projectname

**Reply by daneedev** [reply]
@TheDogHusky 
it is possible, just dont set up a custom domain on the project, that you don't want under the custom domain

**Comment by Jaber-Saed** [comment]
Yes, it's absolutely possible to have a custom domain for just one GitHub Pages project while not affecting the URLs of your existing pages. GitHub Pages allows you to set up a custom domain for a specific repository's Pages site without affecting other repositories.

Here's how you can do it:

1. **Purchase a Domain:**
   - If you haven't already, purchase a custom domain from a domain registrar (e.g., Namecheap, GoDaddy, Google Domains). In your case, it would be "mycustomdomain.org."

2. **Configure DNS Records:**
   - Set up the necessary DNS records for your custom domain to point to GitHub's servers. You'll need to create two DNS records: a CNAME record for your subdomain (e.g., "www") and an A record with the GitHub Pages IP addresses. Refer to GitHub's documentation for the exact IP addresses and DNS setup instructions.

3. **Repository Settings:**
   - In your GitHub repository (the one you want to associate with the custom domain), navigate to the "Settings" tab.

4. **GitHub Pages Settings:**
   - Scroll down to the "GitHub Pages" section.

5. **Custom Domain:**
   - Enter your custom domain (e.g., "mycustomdomain.org") in the "Custom domain" field.

6. **Save:**
   - Save your changes. GitHub will verify the DNS configuration.

Once these steps are completed, your "myname.github.io/myproject" repository will be accessible via your custom domain ("mycustomdomain.org") without affecting the URLs of other Pages projects. Your other existing Pages projects will continue to be accessible via their default URLs.

Keep in mind a few things:

- DNS changes might take some time to propagate across the internet.
- Ensure that your repository's GitHub Pages settings are properly configured and match the DNS records you set up.
- If you have HTTPS enabled for your GitHub Pages site, GitHub will automatically issue a certificate for your custom domain if the DNS settings are correct.
- Double-check your DNS and GitHub Pages settings if you encounter any issues.

Remember, the above steps might slightly vary based on the specific domain registrar and GitHub's platform updates, so it's always a good idea to consult GitHub's official documentation for the most up-to-date instructions.

**Comment by dangrunebaum** [comment]
Thanks for this useful information!
